tasks/selinux.yml

```diff
@@ -24,33 +23,33 @@
 
 # The following tasks only get executed when selinux is in state enforcing, UsePam is 'no' and the ssh_password module is installed.
 # See this issue for more info: https://github.com/hardening-io/ansible-ssh-hardening/issues/23
-- block:
-  - name: Create selinux custom policy drop folder
-    file:
-      path: '{{ ssh_custom_selinux_dir }}'
-      state: 'directory'
-      owner: 'root'
-      group: 'root'
-      mode: '0750'
+- when: (ssh_use_pam|bool) and (ssh_password_module.stdout.find('ssh_password') != 0)
+  block:
+    - name: create selinux custom policy drop folder
+      file:
+        path: '{{ ssh_custom_selinux_dir }}'
+        state: 'directory'
+        owner: 'root'
+        group: 'root'
+        mode: '0750'
 
-  - name: Distributing custom selinux policies
-    copy:
-      src: 'ssh_password'
-      dest: '{{ ssh_custom_selinux_dir }}'
+    - name: distributing custom selinux policies
+      copy:
+        src: 'ssh_password'
+        dest: '{{ ssh_custom_selinux_dir }}'
 
-  - name: check and compile policy
-    command: checkmodule -M -m -o {{ ssh_custom_selinux_dir }}/ssh_password.mod {{ ssh_custom_selinux_dir }}/ssh_password
+    - name: check and compile policy
+      command: checkmodule -M -m -o {{ ssh_custom_selinux_dir }}/ssh_password.mod {{ ssh_custom_selinux_dir }}/ssh_password
 
-  - name: create selinux policy module package
-    command: semodule_package -o {{ ssh_custom_selinux_dir }}/ssh_password.pp  -m {{ ssh_custom_selinux_dir }}/ssh_password.mod
+    - name: create selinux policy module package
+      command: semodule_package -o {{ ssh_custom_selinux_dir }}/ssh_password.pp  -m {{ ssh_custom_selinux_dir }}/ssh_password.mod
 
-  - name: install selinux policy
-    command: semodule -i {{ ssh_custom_selinux_dir }}/ssh_password.pp
+    - name: install selinux policy
+      command: semodule -i {{ ssh_custom_selinux_dir }}/ssh_password.pp
 
-  when: not ssh_use_pam | bool and ssh_password_module.stdout.find('ssh_password') != 0
 
 # The following tasks only get executed when selinux is installed, UsePam is 'yes' and the ssh_password module is installed.
 # See http://danwalsh.livejournal.com/12333.html for more info
 - name: remove selinux-policy when Pam is used, because Allowing sshd to read the shadow file directly is considered a potential security risk
   command: semodule -r ssh_password
-  when: ssh_use_pam | bool and ssh_password_module.stdout.find('ssh_password') == 0
+  when: (ssh_use_pam|bool) and (ssh_password_module.stdout.find('ssh_password') == 0)
```

Comment:
Yes, the comment is wrong. Thoe code still works. I'm in favor of reworking the conditionals (like here https://github.com/dev-sec/ansible-os-hardening/blob/master/tasks/user_accounts.yml#L18-L20) and fixing the comment.

Reply:
I replaced `find` with a more pythonic approach using `in`

Reply:
I really like you solution, now it is much easier to read for me.

But it seems I confused you with my commenting this on the lower task. I think the large block should be executed when the ssh_password module is not installed (since the module is installed in this block). and the single task at the bottom should be executed when the ssh_password module is installed ()so it can be uninstalled). Now it is the other way around. Could you please swap this?

Sorry for the confusion.